lib/engine/step/g_1846/assign_steamboat.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require_relative 'assign'
+
+module Engine
+  module Step
+    module G1846
+      class AssignSteamboat < G1846::Assign
+        ACTIONS = %w[assign pass].freeze
+
+        def actions(entity)
+          blocking_for_steamboat? ? ACTIONS : super
+        end
+
+        def description
+          'Assign Steamboat Company'
+        end
+
+        def assigned_hex
+          @game.hexes.find { |h| h.assigned?(@company.id) }
+        end
+
+        def assigned_corp
+          assignable_corporations.find { |c| c.assigned?(@company.id) }
+        end
+
+        def help
+          assignments = [assigned_hex, assigned_corp].compact.map(&:name)
+
+          targets = []
+          targets << 'hex' if @company.abilities(:assign_hexes)
+          targets << 'corporation or minor' if @company.abilities(:assign_corporation)
+
+          help_text = ["#{@company.owner.name} may assign Steamboat Company to a new #{targets.join(' and/or ')}."]
+          help_text << " Currently assigned to #{assignments.join(' and ')}." if assignments.any?
+
+          help_text
+        end
+
+        def pass_description
+          'Skip Assign'
+        end
+
+        def active_entities
+          @company ? [@company] : super
+        end
+
+        def active?
+          blocking_for_steamboat?
+        end
+
+        def blocking?
+          blocking_for_steamboat? || super
+        end
+
+        def blocks?
+          true
+        end
+
+        def pass!
+          super
+          @round.start_operating
```

Comment:
hm why ?

Reply:
I did some weird stuff to get the steamboat assignment to happen at the beginning of the OR, including changing `Round::Operating#setup` to not call `start_operating`, so I figured `AssignSteamboat` would need to call it. Don't remember if I actually tested it out without this line though.

Reply:
we definitely should not do it like this